File: y2-t3_keep-playing/frontend/lib/models/organiser.dart
import 'user.dart';

class Organiser {
  final List<int> favourites;
  final List<int> blocked;
  final String defaultSport;
  final String defaultRole;
  final String defaultLocation;
  final int? defaultPrice;

  const Organiser({
    required this.favourites,
    required this.blocked,
    required this.defaultSport,
    required this.defaultRole,
    required this.defaultLocation,
    this.defaultPrice,
  });

  factory Organiser.fromJson(Map<String, dynamic> json) => Organiser(
        favourites: (json['favourites'] as List<dynamic>).map((e) => e as int).toList(),
        blocked: (json['blocked'] as List<dynamic>).map((e) => e as int).toList(),
        defaultSport: json['default_sport'] as String? ?? '',
        defaultRole: json['default_role'] as String? ?? '',
        defaultLocation: json['default_location'] as String? ?? '',
        defaultPrice: json['default_price'] as int?,
      );

  bool isFavourite(User user) => favourites.contains(user.pk);
  bool isBlocked(User user) => blocked.contains(user.pk);
}

class OrganiserDefaults {
  final String defaultSport;
  final String defaultRole;
  final String defaultLocation;
  final int? defaultPrice;

  const OrganiserDefaults({
    required this.defaultSport,
    required this.defaultRole,
    required this.defaultLocation,
    this.defaultPrice,
  });

  Map<String, dynamic> toJson() => {
        'default_sport': defaultSport,
        'default_role': defaultRole,
        'default_location': defaultLocation,
        'default_price': defaultPrice,
      };
}

File: y2-t3_keep-playing/backend/app/models/organiser.py
from django.db import models

from .user import User


class Organiser(models.Model):
    user = models.OneToOneField(User, on_delete=models.CASCADE, primary_key=True)
    favourites = models.ManyToManyField(User, related_name='favourite_coaches', blank=True)
    blocked = models.ManyToManyField(User, related_name='blocked_coaches', blank=True)
    default_location = models.CharField(max_length=100, blank=True, default='')
    default_price = models.IntegerField(null=True, blank=True)
    default_sport = models.CharField(max_length=50, blank=True, default='')
    default_role = models.CharField(max_length=50, blank=True, default='')

    def __str__(self):
        return f"Organiser: {self.user.username}"
